src/app/templates/app/archive_requests.html

```diff
@@ -70,14 +71,38 @@
     });
     $('body').on('click', 'button[id^="unarchive_button"]', function (en) {
       en.stopImmediatePropagation();
-      $("#modal-header").addClass("yellow-modal");
-	    $(".modal-footer").html('<button class="btn btn-success" id="confirm_unarchive"><span class="glyphicon glyphicon-off"></span> Confirm</button>');
-      $("#modal-body").html('Are you sure you want to unarchive this license?');
-      $("#myModal").modal({
-        backdrop: 'static',
-        keyboard: true,
-        show: true,
-      })
+      var githubLogin =  "{{ github_login }}" ;
+      if ((githubLogin) != "None"){
```

Comment:
This again has security flaws. User can see this js in the source page for html. Now they can run the same thing on console to add these html elements bypassing the if else check with simple `gitHubLogin = true`. Please see this : https://docs.djangoproject.com/en/3.0/ref/templates/builtins/#if
to introduce this check in html itself.

Reply:
@rtgdk  The function runs whenever the user clicks on the POST button. Without using java script, How will I be able to write the function. or I could instead of declaring the variable just write if ( "{{ github_login }}" != "None"?

Reply:
@anshuldutt21 Have you had a look at https://docs.djangoproject.com/en/3.0/ref/templates/builtins/#if

We can keep the if..else in the html itself. 
Have a look at line: https://github.com/spdx/spdx-online-tools/blob/8e2974f73a412d3c576f874379d5f19adc06bd2f/src/app/templates/app/base.html#L162

You can find the same if..else in any of my django projects.

Reply:
Hi @rtgdk , I apologize for replying too late.
`
	var githubLogin = $("#githubLogin").text();
	/* if user not authenticated using GitHub, display modal with login button */
	if(githubLogin == "False"){
		$("#modal-header").removeClass("red-modal green-modal");
		$("#modal-header").addClass("yellow-modal");
		$(".modal-footer").html('<button class="btn btn-success" id="github_auth_begin"><span class="glyphicon glyphicon-ok"></span> Confirm</button>');
		$("#modal-body").html('To submit a license, you must log in using Github.  You will now be redirected to the github login.  Please allow the requested permissions.  If you do not have a github account, you can <a href="https://github.com/">sign up</a> for free or you can email your new license request to <a href="mailto:spdx-legal@spdx.org">spdx-legal@spdx.org</a>.');
        $("#myModal").modal({
           backdrop: 'static',
           keyboard: true,
           show: true
        });
	}
	}
`
[https://github.com/spdx/spdx-online-tools/blob/master/src/app/templates/app/submit_new_license.html](url)
In the code of submit new license as above, it was done in a similar way because by HTML method we could'nt call the function. The same issue is with the license request.html code. Here we are calling the onclick method of button (archive) and then accordingly we are formatting the html of modal using the add function. 
Also, even if a person is able to edit the javascript he wont be able to archive the request since I have added a check in the backend view that only an authorised person can archive a  request.

Reply:
@anshuldutt21 But the question is, why are we showing the button when user is not logged in or does not have permission - just to show them error dialog box saying "You are not allowed to do this.". Instead, to improve UX we should show the button only when the user is logged in and has perms to do this. This can be done in html if else itself with some backend processing of user info. This would also be easier in terms of the amount of code to be written and will be easily manageable. 
Also, if you find such code pre written like in `submit_new_license.html` feel free to change that too in this PR.